main.go

```diff
@@ -970,13 +1040,17 @@ SAMPLES:
 					totalGtCount += 2
 					totalAltCount += 2
 					homs = append(homs, header[i])
+					dosages = append(dosages, int16(2))
+
 					continue SAMPLES
 				}
 			}
 
 			// N|., .|N, .|., N/., ./N, ./. are all considered missing samples, because if one site is missing, the other is likely unreliable
 			if sampleGenotypeField[0] == '.' || sampleGenotypeField[2] == '.' {
 				missing = append(missing, header[i])
+				dosages = append(dosages, nil)
```

Comment:
Are haploids also considered missing in dosages or is that handled somewhere else?

Reply:
Haploids are taken care of as before, with dosage of 1 (and are called homozygous in the main output)